wd_core.au3

```diff
@@ -1708,6 +1708,7 @@ EndFunc   ;==>__WD_StripPath
 
 Func __WD_ConsoleWrite($sMsg, $iDebugLevel = Default, $iError = @error, $iExtended = @extended)
 	If $iDebugLevel = Default Or $_WD_DEBUG >= $iDebugLevel Then
+		If $iDebugLevel = $_WD_DEBUG_Full Then $sMsg &= ' [' & $iError & '/' & $iExtended & ']'
```

Comment:
Have you testing this change? It isn't working as I would expect. 

I suspect that you need to change `$iDebugLevel` to `$_WD_DEBUG`.

Reply:
With the change mentioned above in place, I now get the following, which matches my expectations --

`_WD_IsLatestRelease ==> Success (0) HTTP status = 200 : True [0/0]`

However, this is what the output from _WD_Startup now looks like --

```
_WD_Startup: OS:	WIN_10 WIN32_NT 22000  [0/6]
_WD_Startup: AutoIt:	3.3.14.5 [0/6]
_WD_Startup: Webdriver UDF:	0.8.1 (Up to date) [0/6]
_WD_Startup: WinHTTP:	1.6.4.2 [0/6]
_WD_Startup: Driver:	geckodriver.exe (64 Bit) [0/6]
_WD_Startup: Params:	--log trace [0/6]
_WD_Startup: Port:	4444 [0/6]
_WD_Startup: Command:	"geckodriver.exe" --log trace  [0/6]
```

How to prevent the additional info from being appended to _WD_Startup output?

Reply:
No.
The intention was to add this suffix only when you use:
```autoit
__WD_ConsoleWrite('........', $_WD_DEBUG_Full)
```

Reply:
> With the change mentioned above in place, I now get the following, which matches my expectations --
> 
> `_WD_IsLatestRelease ==> Success (0) HTTP status = 200 : True [0/0]`
> 

this is duplicated
**_WD_IsLatestRelease ==> Success `(0)` HTTP status = 200 : True `[0/0]`**


> However, this is what the output from _WD_Startup now looks like --
> 
> ```
> _WD_Startup: OS:	WIN_10 WIN32_NT 22000  [0/6]
> _WD_Startup: AutoIt:	3.3.14.5 [0/6]
> _WD_Startup: Webdriver UDF:	0.8.1 (Up to date) [0/6]
> _WD_Startup: WinHTTP:	1.6.4.2 [0/6]
> _WD_Startup: Driver:	geckodriver.exe (64 Bit) [0/6]
> _WD_Startup: Params:	--log trace [0/6]
> _WD_Startup: Port:	4444 [0/6]
> _WD_Startup: Command:	"geckodriver.exe" --log trace  [0/6]
> ```
> 

as I said

> The intention was to add this suffix only when you use:
> 
> ```autoit
> __WD_ConsoleWrite('........', $_WD_DEBUG_Full)
> ```

exactly for this reason which you have pointed, and because of duplication I mention above.

Reply:
I'm having some trouble with the direction this is headed--

- __WD_ConsoleWrite should behave the same no matter the calling routine

- $iDebugLevel was intended to filter output by comparing it to $_WD_DEBUG. So when you have the line `__WD_ConsoleWrite($sFuncName & ":  Some message here", $_WD_DEBUG_Info)`, it should output the same content as long as $_WD_DEBUG is greater than or equal to $iDebugLevel 

- If you want to dynamically alter the message (ie: add error codes), then I'm thinking that should occur in __WD_Error.

Reply:
> this is duplicated
_WD_IsLatestRelease ==> Success (0) HTTP status = 200 : True [0/0]

True

Reply:
> * __WD_ConsoleWrite should behave the same no matter the calling routine

My intention was to have an exception form this rule
```autoit
__WD_ConsoleWrite($sFuncName & ": Some message here", $_WD_DEBUG_Full)
```

You must to remember that `$_WD_DEBUG_Full` was intended to by used for debuging.

> * $iDebugLevel was intended to filter output by comparing it to $_WD_DEBUG. So when you have the line `__WD_ConsoleWrite($sFuncName & ":  Some message here", $_WD_DEBUG_Info)`, it should output the same content as long as $_WD_DEBUG is greater than or equal to $iDebugLevel

yes and in case when `$_WD_DEBUG_Full` is used, as it is for debuging, thus you need to have `Error/Extended` message to easily do the debug stuff.

> * If you want to dynamically alter the message (ie: add error codes), then I'm thinking that should occur in __WD_Error.

currently `__WD_Error` is used only on `Return SetError(__WD_Error(....` point.
And this rule should not be changed.
but as I said few lines above:
**in case when `$_WD_DEBUG_Full` is used, as it is for debuging, thus you need to have `Error/Extended` message to easily do the debug stuff.**

Reply:
Edited previous message few times, sorry. No more edits for this message.

Reply:
> You must to remember that $_WD_DEBUG_Full was intended to be used for debuging.

I think we have slightly different views on this. To me, $_WD_DEBUG is used to control the debugging level. The $_WD_DEBUG_* enums define the available debugging levels.

IMO, the functionality gets too convoluted when you attempt to overload $iDebugLevel and use it for multiple purposes.

Here are some options I can see to avoid this confusion --

- Add another optional parameter that could be used to trigger the extended output
- Add a separate function that would handle the message adjustment and then pass it to __WD_ConsoleWrite

Let me know if you come up with any others.

Reply:
> I think we have slightly different views on this. To me, $_WD_DEBUG is used to control the debugging level. The $_WD_DEBUG_* enums define the available debugging levels.

How it is related to this usage:
```autoit
__WD_ConsoleWrite($sFuncName & ": URL=" & $sURL, $_WD_DEBUG_Info)
```

?

Reply:
> IMO, the functionality gets too convoluted when you attempt to overload $iDebugLevel and use it for multiple purposes.

This functionality IMO is for intentionally use `__WD_ConsoleWrite` out of `__WD_Error` which should differ on current `$_WD_DEBUG` level.
Thus it can differ not only on showing or not showing Message, it can also differ by adding suffix in case of `$_WD_DEBUG_Full` is used.


> * Add another optional parameter that could be used to trigger the extended output

It would compilcate function usage, more then current proposal.

Reply:
> How it is related to this usage:
>
> __WD_ConsoleWrite($sFuncName & ": URL=" & $sURL, $_WD_DEBUG_Info)

$_WD_DEBUG_Info is a debugging level. In the line of code you posted, it defines the minimum debugging level that is required for the console output to occur. This table shows the different debugging levels and if the output would occur --

| $_WD_DEBUG       | Output? |
|------------------|:-------:|
| $_WD_DEBUG_None  |    No   |
| $_WD_DEBUG_Error |    No   |
| $_WD_DEBUG_Info  |   Yes   |
| $_WD_DEBUG_Full  |   Yes   |

Does this answer your question? If not, please elaborate.